packages/platform-server/src/utils.ts

```diff
@@ -187,20 +212,42 @@ export function renderModule<T>(moduleType: Type<T>, options: {
  * @developerPreview
  */
 export function renderApplication<T>(rootComponent: Type<T>, options: {
+  /** @deprecated use `APP_ID` token to set the application ID. */
   appId: string,
   document?: string|Document,
   url?: string,
   providers?: Array<Provider|EnvironmentProviders>,
   platformProviders?: Provider[],
-}): Promise<string> {
-  const {document, url, platformProviders, appId} = options;
+}): Promise<string>;
+export function renderApplication<T>(
+    rootComponentOrBootstrapFn: Type<T>|(() => Promise<ApplicationRef>), options: {
+      appId?: string,
+      document?: string|Document,
+      url?: string,
+      providers?: Array<Provider|EnvironmentProviders>,
+      platformProviders?: Provider[],
+    }): Promise<string> {
+  const {document, url, platformProviders, appId = ''} = options;
   const platform = _getPlatform(platformDynamicServer, {document, url, platformProviders});
+
+  if (isApplicationBootstrapFn(rootComponentOrBootstrapFn)) {
```

Comment:
nit: I'd shorten it a bit (without loosing the meaning/readability):
```suggestion
  if (isBootstrapFn(rootComponentOrBootstrapFn)) {
```